### ISSUE TITLE
Refactor `add_arguments` in `llama_generate` recipe to use updated `TuneArgumentParser`

### ISSUE THREAD

**Issue by RdoubleA** [pytorch/torchtune#281]
After #258, specifying recipe parameters using `add_argument` is no longer needed. Instead, use `TuneArgumentParser`'s methods to parse config and CLI overrides into a dataclass which is passed into the recipe.
- Create a params dataclass for `llama_generate` with appropriate validation logic
- Remove all `add_argument` calls and use `TuneArgumentParser` to funnel parsed arguments into dataclass
- Create an example config to run `llama_generate` recipe
